src/symbolize/symbolizer.rs

```diff
@@ -534,16 +534,26 @@ impl SymbolizeHandler<'_> {
     }
 
     fn handle_elf_addr(&mut self, addr: Addr, file_off: u64, entry_path: &EntryPath) -> Result<()> {
+        let debug_dirs = self
+            .symbolizer
+            .maybe_debug_dirs(self.debug_syms)
+            .map(|dirs| {
+                dirs.iter()
+                    .filter_map(|dir| Some(entry_path.root_path.join(dir.strip_prefix("/").ok()?)))
+                    .chain(dirs.iter().cloned())
+                    .collect::<Vec<_>>()
+            });
+
+
         let resolver = if self.map_files {
-            self.symbolizer.elf_cache.elf_resolver(
-                entry_path,
-                self.symbolizer.maybe_debug_dirs(self.debug_syms),
-            )
+            self.symbolizer
+                .elf_cache
+                .elf_resolver(entry_path, debug_dirs.as_deref())
         } else {
             let path = &entry_path.symbolic_path;
             self.symbolizer
                 .elf_cache
-                .elf_resolver(path, self.symbolizer.maybe_debug_dirs(self.debug_syms))
+                .elf_resolver(path, debug_dirs.as_deref())
```

Comment:
Unfortunately I think the logic is mildly and subtly wrong now: the cache expects the list of debug directories to always be the same, because the "resolver" being looked up will be the same for the same path. However, what can happen now is that the first process has root directory `/` and a subsequent one uses something else, then, because of caching, we wouldn't be checking the new root for debug link targets.

Realistically, it's probably rare and the fix won't be trivial, but let's at least add a TODO about this issue?